### PR TITLE
[SYCL] Enable FTZ operations for CUDA/PTX backend via -fcuda-flush-denormals-to-zero

### PR DESCRIPTION
**Pull request by P-Jensen**
Gives a small performance benefit for Gromacs on NVidia targets

### CODE REVIEW

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 1]
```diff
@@ -3003,6 +3003,12 @@ static void RenderFloatingPointOptions(const ToolChain &TC, const Driver &D,
       RoundingMathPresent = false;
       break;
 
+    case options::OPT_fcuda_flush_denormals_to_zero:
+    case options::OPT_fgpu_flush_denormals_to_zero:
+      DenormalFPMath = llvm::DenormalMode::getPositiveZero();
+      DenormalFP32Math = llvm::DenormalMode::getPositiveZero();
+      break;
```
Comment (mdtoguchi) [inline clang/lib/Driver/ToolChains/Clang.cpp:3010]: Could you add a driver test verifying the expect values when the options are set?

File: clang/lib/CodeGen/CodeGenModule.cpp [hunk 2]
```diff
@@ -815,13 +815,15 @@ void CodeGenModule::Release() {
...
     // Indicate whether __nvvm_reflect should be configured to flush denormal
     // floating point values to 0.  (This corresponds to its "__CUDA_FTZ"
     // property.)
     getModule().addModuleFlag(llvm::Module::Override, "nvvm-reflect-ftz",
-                              CodeGenOpts.FP32DenormalMode.Output !=
-                                  llvm::DenormalMode::IEEE);
+                              (CodeGenOpts.FP32DenormalMode.Output !=
+                                  llvm::DenormalMode::IEEE) ||
+                              (CodeGenOpts.FPDenormalMode.Output !=
+                                  llvm::DenormalMode::IEEE));
```
Comment (premanandrao) [inline clang/lib/CodeGen/CodeGenModule.cpp:826]: Is the logic here that we flush to zero when either of the options is set?

**Comment by bader** [comment]
@andykaylor, FYI.

**Review by mdtoguchi — DISMISSED** [review]
OK for driver

**Comment by againull** [comment]
@premanandrao Please review this PR.

**Comment by elizabethandrews** [comment]
> @premanandrao Please review this PR.

@premanandrao is OOO next week . I see he has reviewed the patch earlier and requested a CodeGen test which hasn't been added yet

**Comment by P-Jensen** [comment]
@elizabethandrews @againull This is my bad.  I'll add a test case next week.

**Comment by P-Jensen** [comment]
@elizabethandrews @againull I just added a CodeGen test case.

**Review by premanandrao — APPROVED** [review]
FE changes look good to me.

**Review by smanna12 — APPROVED** [review]

**Review by mdtoguchi — APPROVED** [review]

**Comment by mdtoguchi** [comment]
@intel/llvm-gatekeepers, this looks ready to go.

**Pull request merged**